components/multiselect/multiselect.tsx

```diff
@@ -1,115 +1,146 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
 import {FormattedMessage} from 'react-intl';
 import ReactSelect from 'react-select';
 
+import {ActionMeta} from 'react-select/src/types';
+import {getOptionValue} from 'react-select/src/builtins';
+import {StateManager} from 'react-select/src/stateManager';
+
 import {Constants, A11yCustomEventTypes} from 'utils/constants.jsx';
 import SaveButton from 'components/save_button.jsx';
 
-import MultiSelectList from './multiselect_list.jsx';
+import MultiSelectList from './multiselect_list';
+
+export type Value = {
+    id: string;
+    label: string;
+    value: string;
+};
+
+export type Props = {
+    ariaLabelRenderer: getOptionValue<Value>;
+    buttonSubmitLoadingText?: JSX.Element | string;
+    buttonSubmitText?: JSX.Element | string;
+    handleAdd: (value: Value) => void;
+    handleDelete: (values: Value[]) => void;
+    handleInput: (input: string, multiselect: MultiSelect) => void;
+    handlePageChange?: (newPage: number, currentPage: number) => void;
+    handleSubmit: (value?: Value[]) => void;
+    loading?: boolean;
+    maxValues?: number;
+    noteText?: JSX.Element;
+    numRemainingText?: JSX.Element;
+    optionRenderer: (
+        option: Value,
+        isSelected: boolean,
+        onAdd: (value: Value) => void
+    ) => void;
+    options: Value[];
```

Comment:
there are a lot of mandatory fields now and there was only `ariaLabelRenderer` before, is there any reason for that? I'm not against a more strict checking, but I want to know the reasoning behind.

Reply:
It appeared to me that these props are actually required by the `react-select` library.  I realize this will kind of cascade to future changes as other components are converted...not sure how best to handle this. My inclination would be to keep the code compatible with the dependency types.